Reimplemented FastRedstoneDust for 1.13

DeadlyMC opened the pull request:
**Note : This fix is by Theosib** 

Updated Patch file for BlockRedstoneWire and enabled carpet rule in CarpetSettings.java

**For the new guys who are seeing this PR:**
This fix improves Redstone Dust performance
And even makes redstone behavior a bit more sensible

gnembon commented:
one question - how much is messes up with saved nbts in case someone flip/flops between this and vanilla server?

Earthcomputer commented:
In terms of piston serialization:
- If someone switches to vanilla before loading a carpet NBT then it loads the same way as vanilla, and suffers from the same bug as vanilla does
- If someone switches to carpet before loading a vanilla NBT then it detects that and loads the same way as vanilla.

In terms of the tile entity order:
- Long-term tile entity (non-block36) order within a chunk is not something that is usually relied upon in vanilla anyway.
- The change to the rule only takes effect once the chunk is reloaded; hotswapping between a HashMap and a LinkedHashMap would have been far too complicated.

DeadlyMC committed:
Added RedstoneWireTurbo

Helps improve redstone dust performance
And makes redstone behaviour more sensible

gnembon commented:
Please use static member for the toggle. using .get('fastRedstone') everywhere with this negates lots of the benefits from it. use boolAccelerate() for that.

DeadlyMC committed:
Simplified Code

Using the boolAccelerate() for the carpet rule now...

DeadlyMC commented:
Done :)

DeadlyMC commented:
Additional Video Showing that FastRedstoneDust works:
https://www.youtube.com/watch?v=jEaF-9tiTmI

gnembon committed:
Merge branch 'master' into RSTurbo
